packages/spanner/src/database.js

```diff
@@ -611,9 +653,44 @@ Database.prototype.runStream = function(query, options) {
     };
   }
 
+  if (reqOpts.types) {
+    var types = {};
+
+    for (var prop in reqOpts.types) {
+      var type = reqOpts.types[prop]
+      var child;
+
+      if (is.object(type)) {
```

Comment:
Well, this was my solution for encountering Arrays since you need to specify that it is both an array and has a single child type.